Pull Request Js2 to main branch

pnordboj opened the pull request:
For review

pnordboj committed:
Fixed CSS errors with bootstrap

pnordboj committed:
Changed from filter to sort

Attempted to add filter for comments, reactions and old to new

pnordboj committed:
Added some documentation to JsDoc

pnordboj committed:
Fixed register function

pnordboj committed:
Solved Register Function

pnordboj committed:
Form Validation for Registration function

pnordboj committed:
Finished Post page, added modal for each button

What's new:

1. Changed Reply, Update and Delete button to call a modal with a form to add wanted text for replies.

The Update button calls a modal with a title and text body input to update title and content.

Delete button opens a modal that wants a confirmation that you want to delete the comment.

2. Filters are still WIP but getting there

pnordboj committed:
Added Readme

Cleaned up code, webpage will not display Create Post or View Post aswell as the search and filter options when they are not logged in!

pnordboj committed:
Filters are Working and bug fixing

Filters are now working
Fixed some small bug issues, where the console shows errors.

Final step is documenting with JsDoc

pnordboj committed:
added documentation to jsdocs

pnordboj committed:
Documentation JSDocs

pnordboj committed:
made the view post show 3 posts per row

pnordboj committed:
Update README.md

pnordboj committed:
Update README.md

martink-rsa commented:
Hi Patrick,

Great job on this assignment! There was only one feature issue that I found which was that registration did not work. It seems there is a validation issue.

❌User with @noroff.no or @stud.noroff.no email can register profile
✔Registered user can login
✔User can view a post content feed
✔User can filter the post content feed
✔User can search the post content feed
✔User can view a post content item by ID
✔User can create a post content item
✔User can update a post content item
✔User can delete a post content item

## Unable to register:

Trying to register has a validation error: 

![image](https://user-images.githubusercontent.com/52622303/195879276-9c6b1439-4a01-4fa3-9b39-294af9723573.png)


## Broken image:

Broken images appear with no `src`. It might be that the post itself doesn’t have an image, but you’re trying to render it. You should conditionally check if an image exists and then display it.

## Incorrect details still try to login:

The app will still try to log users in even when the details are incorrect, which leads to other issues because no user is logged in: 

![image](https://user-images.githubusercontent.com/52622303/195879345-220ad623-e4cf-4b81-8a07-745445e8c0eb.png)


## No log out button:

This isn’t a requirement, but there is no log-out button.

Please don’t hesitate to contact the teachers if you need any assistance with the coursework.

Kind regards,
Martin

pnordboj committed:
Added Logout and Fixed Login and Register Validation

- Added Logout button, which clears local storage and refreshes page.

-Fixed Registration where it dose not recognize and email or a noroff /stud.noroff email

-Login is now using a  login validation, also checks if the user is authorized by checking if the name of the user is undefined or not.

pnordboj committed:
Cleaned up Code

pnordboj committed:
Modified JsDoc index page

Added return to home button

Added all source pages to all page on JsDocs

pnordboj commented:
> Hi Patrick,
> 
> Great job on this assignment! There was only one feature issue that I found which was that registration did not work. It seems there is a validation issue.
> 
> ❌User with @noroff.no or @stud.noroff.no email can register profile ✔Registered user can login ✔User can view a post content feed ✔User can filter the post content feed ✔User can search the post content feed ✔User can view a post content item by ID ✔User can create a post content item ✔User can update a post content item ✔User can delete a post content item
> ## Unable to register:
> 
> Trying to register has a validation error:
> 
> ![image](https://user-images.githubusercontent.com/52622303/195879276-9c6b1439-4a01-4fa3-9b39-294af9723573.png)
> ## Broken image:
> 
> Broken images appear with no `src`. It might be that the post itself doesn’t have an image, but you’re trying to render it. You should conditionally check if an image exists and then display it.
> ## Incorrect details still try to login:
> 
> The app will still try to log users in even when the details are incorrect, which leads to other issues because no user is logged in:
> 
> ![image](https://user-images.githubusercontent.com/52622303/195879345-220ad623-e4cf-4b81-8a07-745445e8c0eb.png)
> ## No log out button:
> 
> This isn’t a requirement, but there is no log-out button.
> 
> Please don’t hesitate to contact the teachers if you need any assistance with the coursework.
> 
> Kind regards, Martin

Registration issue is now solved, Logout button is added, logging in with incorrect details is now solved
